libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt

```diff
@@ -129,10 +148,9 @@ class CarRouteLine(
 
     private fun onRoutesChanged(routes: List<NavigationRoute>) {
         logAndroidAuto("CarRouteLine onRoutesChanged ${routes.size}")
-        mainCarContext.mapboxCarMap.carMapSurface?.getStyle()?.let { style ->
-            if (routes.isNotEmpty()) {
-                val routesMetadata =
-                    mainCarContext.mapboxNavigation.getAlternativeMetadataFor(routes)
+        carMapSurface?.getStyle()?.let { style ->
```

Comment:
We can pass the style from `onAttached`.

Reply:
Needed to take this out of the scope that repeated when the style changed.

Technically we could do something like this
``` kotlin
.combine(styleFlow, this@CarRouteLine::onRoutesChanged).collect()

private fun onRoutesChanged(routes: List<NavigationRoute>, style: Style?): List<NavigationRoute> {
    style?.let { style ->
```

But I didn't like the "transform" function where the routes are returned, and it needs to do a null check anyways

Going with this.
``` kotlin
private fun onRoutesChanged(routes: List<NavigationRoute>) {
    styleFlow.value?.let { style ->
```